### PR TITLE
fix: Add missing AWS region endpoints

### LINKED ISSUE #183: Regions (again) .netcore
Hi, seems that all regions other than us-east-1 are not working for `PutObjectAsync,` and there is no way to set regions in any constructor or parameter.  Am I missing something?  When I try `PutObjectAsync`, I get:

`File Upload Error: Minio API responded with message=The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-east-2'`

### PR DESCRIPTION
**Pull request by poornas**
Fixes #183 

A few of the region specific endpoints were missing on .net sdk. Updating as per http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region

### CODE REVIEW

File: Minio/Regions.cs [hunk 1]
```diff
@@ -29,14 +29,22 @@ private Regions()
         /// <returns>Region corresponding to the endpoint. Default is 'us-east-1'</returns>
         internal static string GetRegion(string endpoint)
         {
-            if (endpoint.EndsWith("s3 - ap - northeast - 1.amazonaws.com"))
+            if (endpoint.EndsWith("s3-ap-northeast-1.amazonaws.com"))
```
Comment (krisis) [inline Minio/Regions.cs]: Can't we get the region from the endpoint string without having to use EndsWith method to pattern match individual regions?

Comment (poornas) [reply Minio/Regions.cs]: yes, updated PR.

**Comment by poornas** [comment]
@krisis  and @vadmeste,this is waiting on your feedback.

**Comment by poornas** [comment]
@harshavardhana and @krishnasrinivas , this is waiting for review

**Pull request merged**